source/common/upstream/outlier_detection_impl.cc

```diff
@@ -329,11 +329,17 @@ void DetectorImpl::checkHostForUneject(HostSharedPtr host, DetectorHostMonitorIm
     return;
   }
 
-  std::chrono::milliseconds base_eject_time =
-      std::chrono::milliseconds(runtime_.snapshot().getInteger(
-          "outlier_detection.base_ejection_time_ms", config_.baseEjectionTimeMs()));
-  ASSERT(monitor->numEjections() > 0);
-  if ((base_eject_time * monitor->numEjections()) <= (now - monitor->lastEjectionTime().value())) {
+  auto num_ejections = monitor->numEjections();
+  ASSERT_OR_LOG_AND(num_ejections > 0, return );
+  auto last_ejection_time = monitor->lastEjectionTime();
+  ASSERT_OR_LOG_AND(last_ejection_time, return );
+  if (now < last_ejection_time.value()) {
```

Comment:
how could this happen?

Reply:
I guess it probably can't assuming last_ejection_time is not corrupt. Changed to a simple ASSERT.